website/app/components/doc/page/header/search/index.js

```diff
@@ -0,0 +1,207 @@
+import { action } from '@ember/object';
+
+import { autocomplete } from '@algolia/autocomplete-js';
+
+import { getItemsFunction } from './parts/getItemsFunction';
+import { templatesHeaderFunction } from './parts/templatesHeaderFunction';
+// import { templatesFooterFunction } from './parts/templatesFooterFunction';
+import { templatesItemFunction } from './parts/templatesItemFunction';
+import { htmlPanelFooter } from './parts/htmlPanelFooter';
+import { htmlPanelNoResults } from './parts/htmlPanelNoResults';
+
+export default class DocAlgoliaSearchComponent extends Component {
+  @tracked isModalOpen = false;
+
+  @action
+  didInsertSearchContainer(element) {
+    const autocompleteInstance = autocomplete({
+      id: 'doc-algolia-search-autocomplete-container', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-id
+      container: element, // 'if necessary you can use an ID eg. `#doc-algolia-search-autocomplete-container',` // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-container
+      // panelContainer: '...' // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-panelcontainer
+      panelPlacement: 'full-width', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-panelplacement
+      placeholder: 'Search', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-placeholder
+      // classNames: {}, // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-classnames
+      // plugins: [algoliaRecentSearchesPlugin], // see: https://www.algolia.com/doc/ui-libraries/autocomplete/guides/adding-recent-searches/
+      openOnFocus: true, // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-openonfocus
+      // debug: true, // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-debug
+      detachedMediaQuery: '', // https://www.algolia.com/doc/ui-libraries/autocomplete/core-concepts/detached-mode/
+      translations: {
+        detachedCancelButtonText: 'Close', // https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-translations
+      },
+      onStateChange: ({ state, prevState }) => {
+        // used to reset the query to an empty state/string when the modal is closed (via "click" or "esc" key)
+        if (!state.isOpen && prevState.isOpen) {
+          autocompleteInstance.setQuery('');
+        }
+      },
+
+      //
+      // GET SOURCES
+      // see: https://www.algolia.com/doc/ui-libraries/autocomplete/api-reference/autocomplete-js/autocomplete/#param-getsources
+      getSources({ query }) {
+        if (query.trim() !== '') {
+          return [
```

Comment:
the `getSources` is used to return three different "queries" to the API (called in a single HTTP request) with different parameters so we can have different types of results (general documentation - aka "content" - and tokens and icons)